doc/man7/password-handling.pod

```diff
@@ -0,0 +1,89 @@
+=pod
+
+=head1 NAME
+
+password handling - How diverse parts of OpenSSL treat pass phrases
+
+=head1 DESCRIPTION
+
+=head2 The general case
+
+In a modern world with all sorts of character encodings, the treatment of pass
+phrases has become increasingly complex.
+
+The OpenSSL library doesn't treat pass phrases in any special way as a general
+rule, and trusts the application or user to choose a suitable character set
+and stick to that throughout the life of affected objects.
+This means that for an object that was encrypted using a pass phrase encoded in
+ISO-8859-1, that object needs to be decrypted using a pass phrase encoded in
+ISO-8859-1.
+Anything else will have the decryption fail.
+
+=head2 PKCS#12
+
+PKCS#12 is a bit different regarding pass phrase encoding.
+The standard stipulates that the pass phrase shall be encoded as a BMPString,
+i.e. UTF-16.
+
+OpenSSL tries to adapt to this requirements in one of the following manners:
+
+=over 4
+
+=item 1.
+
+Tries to read the pass phrase as UTF-8 and re-encode that to UTF-16, or failing
+that...
+
+=item 2.
+
+Assumes that the pass phrase is encoded in ASCII or ISO-8859-X and
+opportunistically prepends each byte with a zero byte and uses the result
+as a BMPString.
+
+=back
+
+Note that historical OpenSSL versions did variant 2 only, and that is the
```

Comment:
Which versions? No fairy tales, version numbers please ;-)

Reply:
Wait, what?  You want me to actually *research*?  :-D

Reply:
It was easier than I though...  1.0.2 does the old way only.  For some reason, I thought 1.0.2 did the UTF-8 dance as well, but I must have looked in the wrong spot.